src/App/Fossa/Analyze.hs

```diff
@@ -194,7 +195,8 @@ analyze basedir destination override unpackArchives filters = do
     FoundSome someProjects -> case destination of
       OutputStdout -> logStdout . pretty . decodeUtf8 . Aeson.encode . buildResult $ NE.toList someProjects
       UploadScan apiOpts metadata -> do
-        revision <- mergeOverride override <$> inferProject (unBaseDir basedir)
+        revision <- mergeOverride override <$> (inferProjectFromVCS basedir <||> inferProjectDefault basedir)
+        saveRevision revision
```

Comment:
The trick is that we use `(inferProjectFromVCS basedir <||> inferProjectDefault basedir)` followed by a `saveRevision` in places where we do analysis, and `(inferProjectFromVCS basedir <||> inferProjectCached basedir <||> inferProjectDefault basedir)` elsewhere